modules/Product/app/Http/Controllers/Api/Web/Panel/V1/ProductController.php

```diff
@@ -26,34 +26,75 @@ public function index(Request $request)
     {
         try {
             $products = $this->productService->index($request);
-        } catch (\Exception $exception) {
+
+            return $this->successWithData($products);
+        } catch (ProductModuleException $exception) {
             return $this->error('Error has been occurred', Response::HTTP_BAD_REQUEST);
+        } catch (\Exception $exception) {
+            return $this->error('An unexpected error occurred. Please try again later.', Response::HTTP_INTERNAL_SERVER_ERROR);
         }
-        return $this->successWithData($products);
     }
 
     public function store(StoreRequest $request)
     {
         try {
             $products = $this->productService->store($request);
+
+            return $this->successWithData($products, Response::HTTP_CREATED);
+        } catch (ProductModuleException $exception) {
+            return $this->error('An error occurred while storing the product.', Response::HTTP_BAD_REQUEST);
         } catch (\Exception $exception) {
-            return $this->error('Error has been occurred', Response::HTTP_BAD_REQUEST);
+            return $this->error('An unexpected error occurred. Please try again later.', Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public function show(Product $product) //TODO ProductResources
+    {
+        try {
+            $productData = $this->productService->show($product);
+
+            return $this->successWithData($productData);
+        } catch (ProductModuleException $exception) {
+            return $this->error('Unable to retrieve product details. Please try again later.', Response::HTTP_NOT_FOUND);
+        } catch (\Exception $exception) {
+            return $this->error('An unexpected error occurred. Please try again later.', Response::HTTP_INTERNAL_SERVER_ERROR);
         }
+    }
+
+    public function update(Request $request, Product $product)
```

Comment:
1 - Please add a FormRequest for update product to check input data .
2 - add try catch